v2/services/authorize/credentialsmanager/manager.go

```diff
@@ -0,0 +1,19 @@
+package credentialsmanager
+
+import "context"
+
+type DataStore struct {
+	CA  []byte `json:"ca"`
+	Key []byte `json:"key"`
+	Crt []byte `json:"crt"`
+}
+
+type CredentialsManager interface {
+	GetDataStore(ctx context.Context, secretsName string) (*DataStore, error)
+}
+
+type CredentialsManagerBuilder struct{}
+
+func New() *CredentialsManagerBuilder {
```

Comment:
Not sure a separate builder type is strictly necessary? The struct to implement the different behavior for AWS SDK v1/v2 can be embedded in `credentialsmanager`, rather than having the builder return different implementations of the `CredentialsManager` interface.

Something like:
```golang
type secretFetcher interface {
  GetSecret(ctx context.Context, secretName string) (DataStore, error)
}

type v1 struct {
  client SMAPIv1
}

func (a *v1) GetSecret(ctx context.Context, secretName string) (DataStore, error) {
  a.client.GetSecretValue(...)
}
```

And then have `UsingSDKV1(sess *session.Session)` just set `c.secretFetcher = &v1{client: secretsmanager.New(sess)}`, or something like that.

Reply:
Yeah that seems a lot easier. I have changed it :)